### DISCUSSION TITLE
Allows options for setValue to prevent setting dirty

### DISCUSSION THREAD

**Discussion by KershSoftware** [react-hook-form/react-hook-form#1894] [Ideas] [answered]
I came across a use case that I could see benefit in modifying `setValue`; instead of taking `shouldValidate` as last param, I think it would make more sense to take an options object of params that contain `shouldValidate` and additional options.

Specifically a desired additional option would be `preventDirty` or whatever name makes sense. But to prevent the controller from updating the `dirtyFieldsRef` and `isDirtyRef` would be a great optional addition as I have ran into a use case where I'm automating select menu options based on another menu's active option. We desired to auto-set specific options based on this dynamic list changing, but also rely on the form's dirty state to be accurate (which is not correct from the user's perspective to be dirty once `setValue` is programmatically called by our logic).

**Comment by bluebill1049** [comment, marked as answer]
hey, @kerosan this is a great idea! please create a feature request I think we still can make it in time for V6. it's a good breaking change to have.

`setValue(xxxx, { shouldDirty = true, shouldValid = false})`

**Reply by bluebill1049** [reply]
This will be in the V6 merged yesterday.
